gradle-plugin/src/test/kotlin/com/google/devtools/ksp/gradle/GradleCompilationTest.kt

```diff
@@ -115,4 +117,117 @@ class GradleCompilationTest {
     }
+
+    @Test
+    fun testCommandLineArgumentProvider() {
+        testRule.setupAppAsAndroidApp()
+        testRule.appModule.addSource("Foo.kt", "class Foo")
+        testRule.appModule.addSource(
+            "Entity.kt",
+            """
+            import androidx.room.Entity
+            import androidx.room.PrimaryKey
+            import androidx.room.ColumnInfo
+            
+            @Entity
+            data class User(
+                @PrimaryKey val uid: Int,
+                @ColumnInfo(name = "first_name") val firstName: String?,
+                @ColumnInfo(name = "last_name") val lastName: String?
+            )
+            """.trimIndent()
+        )
+        testRule.appModule.addSource(
+            "UserDao.kt",
+            """
+            import androidx.room.Dao
+            import androidx.room.Query
+            
+            @Dao
+            interface UserDao {
+                @Query("SELECT * FROM User")
+                fun getAll(): List<User>
+            }
+            """.trimIndent()
+        )
+        testRule.appModule.addSource(
+            "Database.kt",
+            """
+            import androidx.room.Database
+            import androidx.room.RoomDatabase
+            
+            @Database(entities = [User::class], version = 1)
+            abstract class Database : RoomDatabase() {
+                abstract fun userDao(): UserDao
+            }
+            """.trimIndent()
+        )
+        testRule.appModule.dependencies.addAll(
+            listOf(
+                artifact(configuration = "ksp", "androidx.room:room-compiler:2.4.2"),
+                artifact(configuration = "implementation", "androidx.room:room-runtime:2.4.2")
+            )
+        )
+        testRule.appModule.buildFileAdditions.add(
+            """
+                ksp {
+                    arg(Provider(project.layout.projectDirectory.dir("schemas").asFile))
+                }
+                class Provider(roomOutputDir: File) : CommandLineArgumentProvider {
+
+                    @OutputDirectory
+                    val outputDir = roomOutputDir
+
+                    override fun asArguments(): Iterable<String> {
+                        return listOf(
+                            "room.schemaLocation=${'$'}{outputDir.path}"
+                        )
+                    }
+                }
+                tasks.withType<com.google.devtools.ksp.gradle.KspTask>().configureEach { 
+                    doFirst {
+                        options.get().forEach { option ->
+                            println("${'$'}{option.key}=${'$'}{option.value}")
+                        }
+                    }
+                }
+                
+            """.trimIndent()
+        )
+        val result = testRule.runner().withArguments(":app:assembleDebug").build()
+        assertThat(result.output)
+            .contains("apoption=room.schemaLocation=/private${testRule.appModule.moduleRoot}/schemas")
```

Comment:
The location seems platform dependent. The error message is

```
    expected to contain:
        apoption=room.schemaLocation=/private/tmp/junit16660779885054734687/junit9088641825282887015/app/schemas
    but was:
        ...
        apoption=room.schemaLocation=/tmp/junit16660779885054734687/junit9088641825282887015/app/schemas
```

Can you use a regexp to match it? E.g.,

```
    val pattern1 = Regex.escape("apoption=room.schemaLocation=")
    val pattern2 = Regex.escape("${testRule.appModule.moduleRoot}/schemas")
    assertTrue(Regex("$pattern1\\S*$pattern2").containsMatchIn(result.output))
```